backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/database/service/SpecificationDatabaseService.kt

```diff
@@ -13,17 +14,32 @@ import javax.transaction.Transactional
 @Service
 class SpecificationDatabaseService constructor(
     private val specificationRepository: SpecificationRepository,
-    private val entityManager: EntityManager
+    private val entityManager: EntityManager,
+    private val specificationEntityMapper: SpecificationEntityMapper
 ) : SpecificationPersistenceService {
 
     override fun save(specification: Specification) {
-        specificationRepository.save(SpecificationEntity.fromDomain(specification))
+        val specificationEntity = specificationEntityMapper.fromDomain(specification)
+
+        if (specificationRepository.existsByTitle(specificationEntity.title)) {
+            val existingEntity = specificationRepository.findByTitle(specificationEntity.title)
+
+            val versions = existingEntity.versions + specificationEntity.versions
```

Comment:
So if the new version already exists, it is overwritten? I'm not sure if we should prohibit that.

Reply:
What would your expected response be?
- Version is overwritten
- Version exists twice
- Return an error message?

Reply:
Either an outright error, or overwriting is permitted via a configuration option.

My main point is that you don't want to overwrite specs that other people depend on - just like Maven's Nexus. So, ideally, final versions cannot be changed (or only if the admin explicitely allows this), but SNAPSHOTs can. Since we do not yet have a concept of SNAPSHOTs, I'd prefer no overwriting ever.

Reply:
I think we should go with the outright error then.